docs/source/config/details.rst

```diff
@@ -228,3 +228,45 @@ a :ref:`startup file <startup_files>`::
 For more information on filters and what you can do with the ``event`` object,
 `see the prompt_toolkit docs
 <http://python-prompt-toolkit.readthedocs.io/en/latest/pages/building_prompts.html#adding-custom-key-bindings>`__.
+
+
+Enter to execute
+----------------
+
+In the Terminal IPython shell – which by default use the ``prompt_toolkit``
+interface, the semantic of pressing the :kbd:`Enter` key can be ambiguous, in
+some case :kbd:`Enter` should execute code, and in others it should add a new
+line. IPython is using some heuristics to decide whether to execute or insert a
+new line at cursor position. For example, if we detect that the current code is
+not valid Python, then the user is likely editing code and the right behavior is
+to likely to insert a new line. If the current code is a simple statement like
+`ord('*')`, then the right behavior is likely to execute. Though the exact
+desired semantics often varies from users to users.
+
+As the exact behavior of :kbd:`Enter` is subject to disagreement, it has been
+special cased in order for users to completely configure the behavior they like.
+Hence you can have enter to simply alway execute code, if you prefer more fancy
```

Comment:
can have enter always execute code. If you prefer fancier behavior, you need to get your hands dirty and read the ``prompt_toolkit`` and IPython documentation though.